services/sinner/fetcher/scribe.go

```diff
@@ -0,0 +1,206 @@
+// Package fetcher gets data from scribe.
+package fetcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/synapsecns/sanguine/core/metrics"
+	"github.com/synapsecns/sanguine/services/sinner/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/jpillora/backoff"
+	"github.com/synapsecns/sanguine/services/scribe/graphql"
+	"github.com/synapsecns/sanguine/services/sinner/fetcher/client"
+	"github.com/synapsecns/sanguine/services/sinner/logger"
+)
+
+// ScribeFetcher is the interface for fetching events. It uses GQL.
+type ScribeFetcher interface {
+	// FetchLastIndexed fetches the last indexed block per contract.
+	FetchLastIndexed(ctx context.Context, chainID uint32, contractAddress string) (uint64, error)
+	// FetchLogsInRange fetches logs in a range with the GQL client.
+	FetchLogsInRange(ctx context.Context, chainID uint32, startBlock, endBlock uint64, contractAddress common.Address) ([]ethTypes.Log, error)
+	// FetchBlockTime fetches the timestamp of a block.
+	FetchBlockTime(ctx context.Context, chainID int, blockNumber int) (*int, error)
+	// FetchTx fetches the transaction.
+	FetchTx(ctx context.Context, tx string, chainID int, blockNumber int) (*uint64, *string, error)
+	// FetchTxsInRange fetches transactions in a range.
+	FetchTxsInRange(ctx context.Context, chainID uint32, startBlock uint64, endBlock uint64) ([]types.TxSupplementalInfo, error)
+}
+
+type scribeFetcherImpl struct {
+	underlyingClient *client.Client
+	handler          metrics.Handler
+}
+
+const retryThreshold = 5
+
+// NewFetcher creates a new fetcher.
+func NewFetcher(fetchClient *client.Client, handler metrics.Handler) ScribeFetcher {
```

Comment:
Is there any reason to duplicate this from explorer vs just splitting this out into scribe/client?